metricpublisher/schemas/log_event.json

```diff
@@ -0,0 +1,129 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Schema for the log_event lambda request object",
+    "type": "object",
+    "properties": {
+        "request_id": {
+            "type": "string"
+        },
+        "metric_data": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "MetricName": {
+                        "type": "string"
+                    },
+                    "Dimensions": {
+                        "type": "array",
+                        "properties": {
```

Comment:
This is wrong. For type array, you should have something like this: 
```
"type": "array",
  "items": {
    "type": "object",
    "properties": {
```
So the other problem is that your tests did not help catch this.